### PR TITLE
[WIP][DNM] Ports Tranquillite from Paradise, finally bringing equality to mimes everywhere

### PR DESCRIPTION
**Pull request by Buggy123**
Ports Tranquillite from https://github.com/ParadiseSS13/Paradise/pull/3767, which was itself a port/improvement of my original Silencium PR #14870.

Tranquillite is a mime counterpart to Bananium. It can be used to build Reticences, the silent and nigh-invisible mime mech, as well as nearly invisible walls, and floors which nullify all step sounds(even clown shoes).

The way to obtain it is as-of-yet undecided but I was toying with the idea of putting some on the syndicate listening outpost.

Done entirely in webeditor so get ready for a fuckton of commits to be squashed later.

TODO:
 * Get to something that isn't webeditor so i can update my old-as-fuck master (it doesn't even have techwebs).
 * Actually figure out how the fuck you make a real, transparent closed turf this time.
 * Finish porting the silent walking part of the code
 * Fix the two hundred thousand minor little material lists I missed
 * See if its possible to mute all sound on Tranquillite floors a-la low pressure enviroments.

image

### CODE REVIEW

**Comment by Buggy123** [comment]
This is why you keep your master updated, kids.

**Comment by imsxz** [comment]
coming soon: smelting mime ore causes massive lag for the server

**Comment by Buggy123** [comment]
>coming soon: smelting mime ore causes massive lag for the server

goes without saying this will have countless problems with it right now, I can't begin on this proper until I update the branch. Don't even bother pointing out small issues until then TBH, not much point.

**Comment by ExcessiveUseOfCobblestone** [comment]
how about you stop using the webbitor for a 22 file touching port...

**Comment by Buggy123** [comment]
actually its probably gonna be like 30-40 once i get around to everything

**Comment by ExcessiveUseOfCobblestone** [comment]
*sigh*

**Comment by Buggy123** [comment]
>Add more commits by pushing to the BUGGY-MAKES-ANOTHER-SHIT-PR branch on Buggy123/-tg-station.

I mean, you _were_ warned.

**Comment by 81Denton** [comment]
>web editor

Every day we stray further from god's light
By the way, where will tranquilite be available? Lavaland/spess ruins like bananium?

**Comment by uraniummeltdown** [comment]
I would map in a few crates of Tranquillite in the Clown ruin, as if the Clowns stole crates of it

**Comment by 81Denton** [comment]
That sounds good. How does Yiffstation handle it?

**Comment by ExcessiveUseOfCobblestone** [comment]
you could always add a cosmic theatre ruin or something, but that would require more than a webbit 🙄

**Comment by vuonojenmustaturska** [comment]
Before you go any further, i'm just going to point out that it would be a lot easier to re-do this feature from scratch using current master as a base than to spend hours doing web-editor conflict resolution and then wondering after hours of work why it doesn't work even though it compiles. Ores in particular are now sheets and have a better crafting thing than the old crap.

**Comment by Spaceinaba** [comment]
when you reach the point where name your branch after how shit you are at coding maybe you should stop opening prs and start learning to code
other than that I give it a neat/10

**Comment by Xhuis** [comment]
It's a neat idea, but this is a massive PR that cannot and should not be done in web editor.

**Comment by Buggy123** [comment]
Oh right that happens if you push master straight to a branch lets fix that.

**Comment by Xhuis** [comment]
Someone needs to take away your computer and get you help.

**Comment by Buggy123** [comment]
Probably yeah.

In my defense, is dreammaker really that much better than a webeditor? _Really?_

**Comment by Buggy123** [comment]
squash #1 complete.

**Comment by Xhuis** [comment]
Yes, Dream Maker is extremely better than web editor.

**Comment by ExcessiveUseOfCobblestone** [comment]
> is dreammaker really that much better than a webeditor? Really?

Yes else we wouldn't scoff at you using the webeditor.

**Comment by outbools** [comment]
>In my defense, is dreammaker really that much better than a webeditor? Really?
Did you even heard about ATOM which all sane coders use? Install some shit on your computer already, web boi. You can't get away with only browser installed.

**Comment by outbools** [comment]
oh wait ATOM is Electron app so it's a browser too

**Comment by outbools** [comment]
god I hate web development

**Comment by ExcessiveUseOfCobblestone** [comment]
just use DM it's not like it's intensive (byond knows their audience)

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 1]
```diff
@@ -259,6 +260,28 @@ GLOBAL_LIST_INIT(bananium_recipes, list ( \
...
 	. = ..()
 
+/*
+ * Mime
+ */
+/obj/item/stack/sheet/mineral/tranquillite
+	name = "tranquillite"
+	icon_state = "sheet-tranquillite"
+	item_state = "sheet-tranquillite"
+	singular_name = "tranquillite sheet"
```
Comment (uraniummeltdown) [inline code/game/objects/items/stacks/sheets/mineral.dm]: call them berets like on Para, not sheets

Comment (Buggy123) [reply code/game/objects/items/stacks/sheets/mineral.dm]: alright, i'll do that.

**Comment by evsey9** [comment]
fucking USE DREAMMAKER

**Comment by Bawhoppen** [comment]
When I statyed wanting to contribute, I was told there was no such thing as the web editor, and that git desktop was required to make changes.
 It was the best lie anyone ever told me.

**Comment by Buggy123** [comment]
Okay honestly, I don't get it.

Dreammaker:

1: lets you find code issues more quickly by compiling instead of waiting for travis

2: results in less commits

3: has a marginally better interface for navigating the file tree

1 isn't that big of a deal for me, and 3 is only a minor benefit. 2 would seem likes a good idea because you don't have 500,000 commits, which is bad. But you don't have 500,000 commits getting merged, you briefly have 500,000 commits hidden inside this PR until they get squashed. What's the difference?

**Comment by Xhuis** [comment]
Dream Maker also lets you TEST THE CODE. Holy *shit,* Most developers do not generally work in an independent environment for the programs they're developing and then redownload the source code every time they need to test it. There is a *REASON* there are local and remote copies of the repository.

**Comment by Qustinnus** [comment]
Lmao you such a fucking chumbis buggy, using the web editor means youre going to make a fuckton of mistakes, just fucking get atom or visual studio and stop being a fucking mongoloid.

**Comment by Buggy123** [comment]
A large portion of my coding is done when i can't use anything except a Chromebook, tell me how to run byond on this:

And i will.

**Comment by 81Denton** [comment]
Use notepad++, atom or someshit and then compile with dreammaker you chumbis
It‘s not rocket science

**Comment by Buggy123** [comment]
Chromebooks can't run notepad++ or atom.

**Comment by Qustinnus** [comment]
Imagine using the web editor as your main IDE and buying a chrome book.

**Comment by 81Denton** [comment]
>Chromebooks can't run notepad++ or atom.

Oh shit I'm sorry

**Comment by ExcessiveUseOfCobblestone** [comment]
how do you play the game then?

**Comment by Buggy123** [comment]
Mostly during friday/saturday/sunday.

My schedule is essentially this:

Mon-thurs
College(+commute): 6:15 A.M. - 5 P.M. (75% of this is free time but with only a chromebook)
All the other hours: sleep and/or playing video games while trying not to think about homework and/or existential dread and trying not to think about how the fundament of life is pointless drudgery

fri-sun
(12 A.M. - 5 A.M.) - (12 P.M. - 3 P.M.): sleep
all the other hours:playing video games while trying not to think about the upcoming work I have to do, and/or existential dread, drudgery, icing, etc

**Comment by PKPenguin321** [comment]
install linux onto your chromebook :V

**Comment by Buggy123** [comment]
It's been a while since i've checked but I thought the linux version of byond was _only_ dreamdaemon. No dreammaker.

**Comment by ExcessiveUseOfCobblestone** [comment]
how does that even work...

**Comment by Buggy123** [comment]
I just checked and it says

>Development/hosting platform only: cannot be used to play games.

So maybe it can run dreammaker?

**Comment by ExcessiveUseOfCobblestone** [comment]
if you have daemon you can get dm

**Comment by PKPenguin321** [comment]
daemon is a server hosting tool

**Comment by Buggy123** [comment]
image
oh fine ill work on it

**Comment by Buggy123** [comment]
gonna remake it cause its stale

**Pull request closed without merging**